test/powershell/Modules/Microsoft.PowerShell.Utility/NewTemporaryFile.Tests.ps1

```diff
@@ -6,25 +6,57 @@
 <#
     Purpose:
         Verify that New-TemporaryFile creates a temporary file.
+        It has an 'Extension' parameter to change the default extension '.tmp'.
 
     Action:
         Run New-TemporaryFile.
 
     Expected Result:
-        A FileInfo object for the temporary file is returned.
+        A FileInfo object for the temporary file is returned and the temporary file gets created correctly.
 #>
 
 Describe "NewTemporaryFile" -Tags "CI" {
 
-    It "creates a new temporary file" {
-        $tempFile = New-TemporaryFile
+    $defaultExtension = '.tmp'
+
+    AfterEach {
+        if ($null -ne $script:tempFile)
+        {
+            Remove-Item $script:tempFile -ErrorAction SilentlyContinue -Force # variable needs script scope because it gets defined in It block
+        }
+    }
 
-        Test-Path $tempFile | Should be $true
+    It "creates a new temporary file" {
+        $script:tempFile = New-TemporaryFile
+        $tempFile | Should Exist
         $tempFile | Should BeOfType System.IO.FileInfo
+        $tempFile.Extension | Should be $defaultExtension
+    }
+
+    $extensionParameterTestCases = @(
+        @{ extension = 'csv' }
+        @{ extension = '.csv' }
+        @{ extension = '..csv' }
+        @{ extension = 'txt.csv' }
+        @{ extension = '.txt.csv' }
+    )
+    It "creates a new temporary file with a specific extension" -TestCases $extensionParameterTestCases -Test {
+        Param ([string]$extension)
+        
+        $script:tempFile = New-TemporaryFile -Extension $extension
+            $tempFile | Should Exist
+            # Because the internal algorithm does renaming it is worthwhile checking that the original file does not get left behin
+            [System.IO.Path]::ChangeExtension($tempFile, $defaultExtension) | Should Not Exist
+            $tempFile | Should BeOfType System.IO.FileInfo
+            $tempFile.Extension | Should be ".csv"
+            $tempFile.FullName.EndsWith($extension) | Should be $true
+    }
 
-        if(Test-Path $tempFile)
+    It "New-TemporaryItem with an an invalid character in the -Extension parameter should throw NewTemporaryInvalidArgument error" {
+        $invalidFileNameChars = [System.IO.Path]::GetInvalidFileNameChars()
+        foreach($invalidFileNameChar in $invalidFileNameChars)
         {
-            Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
+            { New-TemporaryFile -Extension $invalidFileNameChar -ErrorAction Stop } | ShouldBeErrorId "NewTemporaryInvalidArgument,Microsoft.PowerShell.Commands.NewTemporaryFileCommand"
```

Comment:
Doesn't make sense to test for all invalid chars - enough of one - we should check only the error id.